RELEASE-NOTES.md

```diff
@@ -16,6 +16,30 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
 ### Changed
 - `BLT_C_FILE_EXTS` updated to include `.cuh`
 - Fold `BLT_CLANG_HIP_ARCH` into the `CMAKE_HIP_ARCHITECTURES` variable
+- Source code filename extensions are now detected using regular
+  expressions instead of checking whether the output of
+  `get_filename_component(... ... EXT)` is in a list of filename
+  extensions for each programming language (e.g.,
+  `BLT_C_FILE_EXTS`). CMake's `get_filename_component(... ... EXT)`
+  returns from its filename input a string containing the leftmost
+  period of the filename and all characters to its right (stripping
+  all directory prefixes, assuming use of a left-to-right-script-based
+  character set). This behavior causes problems in filename extension
+  detection with files that have names like `1d.cube.order2.c` because
+  the `get_filename_component` call will output `.cube.order2.c`
+  instead of `.c`.
+
+  While users could add extensions like `.cube.order2.c` to BLT
+  filename extension lists like `BLT_C_FILE_EXTS`, if many such
+  "custom" extensions exist, adding each such extension could be
+  cumbersome. Though CMake 3.14 and later support using
+  `get_filename_component(... ... LAST_EXT)`, earlier versions of
+  CMake do not, and BLT aims to support CMake 3.8+. Instead, regular
+  expressions are used to detect that the filename extension of
+  `1d.cube.order2.c` is `.c`. The regular-expression-based
+  extension-matching implementation continues to support custom
+  filename extensions, and should be backward-compatible with earlier
+  versions of BLT.
```

Comment:
This is a really nice overview of the problem @goxberry

Having said that, I think most of it should be moved elsewhere since the `RELEASE_NOTES` are typically a brief note about what changed.

Consider moving this to the `Fixed` section (since this is a bugfix/improvement)
and modify the writeup to something like the following:
```
- Source code filename extensions now only consider the substring containing the last period, e.g. `1d.cube.order2.c` is considered a `.c` file.
```

Perhaps the current writeup can be used for user docs (in an `.rst` file), or as dev docs as an extended cmake comment)